### ISSUE TITLE
Qt Error when debugging 

### ISSUE THREAD

**Issue by bentsherman** [SystemsGenetics/ACE#67]
I am trying the following script in order to acquire debug information for the GPU error I'm encountering:
```
kinc settings set opencl 0:0
kinc settings set threads 1
kinc settings set logging on

kinc run similarity --input data/Yeast.emx --ccm data/Yeast.ccm --cmx data/Yeast.cmx --clusmethod gmm --corrmethod pearson &

sleep 1
acelog -dt 0 localhost:40000

wait
```

It works at first, but then I get this message 20 times and the logger hangs:
```
QSocketNotifier: Socket notifiers cannot be enabled or disabled from another thread
```

Based on the debug log, this happens at some point after making the opencl worker and the work blocks. It doesn't happen for serial, so I think the issue has to do with the fact that ACE spawns each OpenCL worker in a separate thread, and maybe both the main thread and the worker thread are trying to write to the logger? I figured this couldn't happen since `acelog` is supposed to listen to only one thread.

**Comment by 4ctrl-alt-del** [comment]
This began quite the odyssey for me today.

It started with me wondering why this does not show up on my tests. I discovered my system apparently no longer had any OpenCL devices so was always doing serial.

7 hours and immense frustration later I finally fixed it. I hate Nvidia Corp. Apparently they decided to add more linux kernel dependencies required for their drivers to function with OpenCL without telling anyone. Part of my troubleshooting before that was physically swapping out graphics cards.

Anyway, after that horrific nightmare in the last hour I was able to replicate your bug easily. After looking at the code I realized I have made rather big design flaws that makes its current design unable to cope with other threads. So I am going to have to a total redesign and recoding of the debug system.

That will not start until next week, as I got to spend this entire day bashing my head into a brick wall wondering why opencl vanished on my system.

**Comment by 4ctrl-alt-del** [comment]
OK I believe I have resolved the errors you were getting. I tested it on my end with OpenCL and therefore threading enabled and it works without any complaints. Please let me know if it works for you. The fix is in commit 611a8f02b9aadc94bc13070be12c8883321bb6db.

**Comment by bentsherman** [comment]
Sorry to hear about all of that... you fixed my problem though! Thanks
